### PR TITLE
[WIP] First draft

### PR DESCRIPTION
**Pull request by kjappelbaum**
This is a first draft, I will still need to work on parsing some more metadata (like the specs of the instrument). 

Sorry for the delay, just got my MacBook back from repair last week.

### CODE REVIEW

File: src/__tests__/reader.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { readFileSync } from 'fs';
+import { join } from 'path';
+
+import { readBRML } from '../reader/reader';
+
+describe('readBRML', () => {
+  const data = readFileSync(join(__dirname, '../../data/test.brml'));
+
+  it('check the output dictionary', async () => {
+    let result = await readBRML(data);
```
Comment (lpatiny) [inline src/__tests__/reader.test.js:10]: Please also check some values. At worse you can use https://jestjs.io/docs/en/snapshot-testing

File: src/reader/reader.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import JSZip from 'jszip';
+import { parseDiffractogram } from '../parser/parser';
+
+/**
+ * @param  {} binary
+ * @param  {} options={}
+ */
+// eslint-disable-next-line no-unused-vars
+export function readBRML(binary, options = {}) {
```
Comment (lpatiny) [inline src/reader/reader.js]: Please use rather async / await syntax. We find the code easier to read. It looks like sync code while still being async.

**Review by lpatiny — CHANGES_REQUESTED** [review]
Looks great thanks !

**Comment by kjappelbaum** [comment]
Thanks for the review and the time, I hope I fixed the issues.

File: src/__tests__/reader.test.js (hunk 1 above)
Comment (kjappelbaum) [reply src/__tests__/reader.test.js:10]: Tests on values added in https://github.com/cheminfo/xrd/pull/3/commits/da0378de81d8d6650c1fed400834b62e3722d7f7.

Comment (kjappelbaum) [reply src/__tests__/reader.test.js:10]: Will also move them to the refactored version (https://github.com/kjappelbaum/xrd/tree/spectrum_class).

File: src/reader/reader.js (hunk 2 above)
Comment (kjappelbaum) [reply src/reader/reader.js]: Hopefully a bit clearer in https://github.com/cheminfo/xrd/pull/3/commits/da0378de81d8d6650c1fed400834b62e3722d7f7.

**Pull request merged**